### DISCUSSION TITLE
convert with no options, not a no-op?

### DISCUSSION THREAD

**Discussion by Joshua-Schroijen** [ImageMagick/ImageMagick#4983] [General] [answered]
Hello everyone,  

I have a basic question about ImageMagick's convert command.  
I have noticed that running the command  
`convert image1.jpg image2.jpg`  
yields an image2.jpg that has a different byte content than image1.jpeg, even if image1.jpg is not corrupted as per identify -verboe.

I know that `convert image1.jpg image2.jpg` is handy to fix some corrupted images but what exactly does convert do when you give it no specific options/commands?
I need to know because this corruption fixing effect would be good for my image processing pipeline, but in the context of my technical application knowing exactly what I'm doing is important.  

Thank you in advance,  
Joshua

**Comment by fmw42** [comment, marked as answer]
Convert simply copies the input to the output. But your issue is that for JPG it is lossy compressed and you cannot get back the exact input. Each time you compress and decompress, you lose information. You can set your output -quality to 100, but it is still lossy. For other formats that are not compressed or have lossless compression, the output would be the same as the input ( GIF, PNG, many TIFF compressions, JP2)
